### PR TITLE
Attacker goals

### PR DESCRIPTION
**Pull request by mnm678**
This reopens #20 on the new branch.

### CODE REVIEW

**Review by SteveLasker — COMMENTED** [review]
General comment to refer to artifacts, and not just images as we'll want to secure Helm, WASM, Singularity and others.

File: threatmodel.md [hunk 1]
```diff
@@ -10,3 +10,10 @@ It is assumed that an attacker may perform one or more the following actions:
 
 While it is not always possible to protect against all scenarios, the system should to the extent possible mitigate and/or reduce the damage caused by a successful attack, detect the occurrence of an attack and notify appropriate parties, yet remain usable for parties operating the system.  Furthermore, the system should recover from successful attacks in a way that presents low operational overhead and risk to users.
 
+Attacker Goals:
+1. Trying to have a party install a malicious image under the attackers control.
```
Comment (sudo-bmitch) [inline threatmodel.md]: I'd change "install" to "deploy" to better align with container terminology.

File: threatmodel.md [hunk 2]
```diff
@@ -10,3 +10,10 @@ It is assumed that an attacker may perform one or more the following actions:
...
+2. Trying to have a party install an outdated image.  For example, one with known security vulnerabilities.
```
Comment (sudo-bmitch) [inline threatmodel.md:15]: Change "an outdated image" to "an image who's signature has been revoked by the signer" or something to that effect. That avoids us defining what outdated is, leaving the definition up to the signer to determine and implement according to their own policies.

File: threatmodel.md [hunk 3]
```diff
@@ -10,3 +10,10 @@ It is assumed that an attacker may perform one or more the following actions:
...
+3. Making images unavailable for installation.
```
Comment (sudo-bmitch) [inline threatmodel.md:16]: For Notary, we could phrase this "Disrupt the verification of image signatures."

File: threatmodel.md [hunk 4]
```diff
@@ -10,3 +10,10 @@ It is assumed that an attacker may perform one or more the following actions:
...
+4. Preventing a party from learning about updates to currently installed images.
+5. Convincing a party to download large amounts of data that interfere with the party's system.
```
Comment (sudo-bmitch) [inline threatmodel.md]: To scope this to Notary, we could say something like "Inject a malicious (oversized) signature that disrupts verification" (my phrasing could use improvement). The manifest itself will have a size limit imposed by many runtimes (like containerd), see [this OCI issue](https://github.com/opencontainers/distribution-spec/issues/260) for more context. The manifest includes descriptors that provide both the digest and size of the blob, so if the manifest digest is signed and verified, the only gap for us to cover is the signature artifact itself.

**Review by sudo-bmitch — CHANGES_REQUESTED** [review]
Thanks for capturing these @mnm678. I've done the best I can to rephrase things to align with Notary/Container terms, but please feel free to improve on my wording.

File: threatmodel.md (hunk 1 above)
Comment (mnm678) [inline threatmodel.md]:
```suggestion
1. To have a party deploy a malicious artifact under the attacker's control.
```

File: threatmodel.md (hunk 2 above)
Comment (mnm678) [inline threatmodel.md:15]:
```suggestion
2. To have a party deploy an artifact using a digest or tag that does not have a currently valid signature.  For example, a previous version of an artifact with a signature that was revoked by the signer, or an old version associated with a tag.
```

File: threatmodel.md (hunk 3 above)
Comment (mnm678) [inline threatmodel.md:16]:
```suggestion
3. Disrupt the verification of artifact signatures, for example by making the current version of metadata unavailable.
```

File: threatmodel.md [hunk 5]
```diff
@@ -10,3 +10,10 @@ It is assumed that an attacker may perform one or more the following actions:
 
 While it is not always possible to protect against all scenarios, the system should to the extent possible mitigate and/or reduce the damage caused by a successful attack, detect the occurrence of an attack and notify appropriate parties, yet remain usable for parties operating the system.  Furthermore, the system should recover from successful attacks in a way that presents low operational overhead and risk to users.
 
+Attacker Goals:
+1. Trying to have a party install a malicious image under the attackers control.
+2. Trying to have a party install an outdated image.  For example, one with known security vulnerabilities.
+3. Making images unavailable for installation.
+4. Preventing a party from learning about updates to currently installed images.
```
Comment (mnm678) [inline threatmodel.md]:
```suggestion
4. Prevent a party from learning about updates to currently installed artifacts.
```

File: threatmodel.md (hunk 4 above)
Comment (mnm678) [inline threatmodel.md]:
```suggestion
5. Convince a party to download large amounts of data, such as signatures or metadata, that interfere with the party's system.
```

File: threatmodel.md [hunk 6]
```diff
@@ -10,3 +10,10 @@ It is assumed that an attacker may perform one or more the following actions:
...
+5. Convincing a party to download large amounts of data that interfere with the party's system.
+6. Enabling future attacks of the above types to be carried out more easily.  For example, by causing a party to trust the attacker's key.
```
Comment (mnm678) [inline threatmodel.md]:
```suggestion
6. Enable future attacks of the above types to be carried out more easily.  For example, by causing a party to trust the attacker's key.

## Out of Scope
The following attacks are considered out of scope for Notary v2:
1. Denial of Service (DoS) attacks.
2. Registry validation. A registry may choose to do validation when artifacts are uploaded, but this validation is out of scope of Notary v2.
```

**Review by mnm678 — COMMENTED** [review]
Thanks for the suggestions @sudo-bmitch. I can't edit this directly, but I left some new wording for @JustinCappos to review, or I can re-open this pr from my fork.

File: threatmodel.md (hunk 3 above)
Comment (SteveLasker) [reply threatmodel.md:16]: ~~images~~ - artifacts

Comment (SteveLasker) [reply threatmodel.md:16]: ~~image~~ - artifact

File: threatmodel.md (hunk 4 above)
Comment (SteveLasker) [reply threatmodel.md]: We had a good discussion about size attacks, and I thought we decided to scope this out of Notary. Registries already have mechanisms to cope with DOS attacks, and the descriptor already has enough info for artifact clients to gate these. 
Suggest we remove # 5.

Comment (SteveLasker) [reply threatmodel.md]: Suggest we remove references to "large amounts of data"

File: threatmodel.md (hunk 6 above)
Comment (SteveLasker) [reply threatmodel.md]: I wouldn't say registry validation is out of scope. There's some good ideas here. What I think we said was we'd focus on how notary would validate. In most cases, we see validation happening no the consumption of content from a registry.
But, there was some interesting ideas where a registry might validate in upload. It's not a notary goal, rather an product optiomiation potential. As long as notary can validate things, it's up to the various clients and products to implement that validation at any point they'd like.

File: threatmodel.md (hunk 2 above)
Comment (JustinCappos) [reply threatmodel.md:15]: This changes the meaning a bit.  The prior also would include artifacts that were not the latest (when the latest is requested), but this seems to imply those are fine.

File: threatmodel.md (hunk 3 above)
Comment (JustinCappos) [reply threatmodel.md:16]: This is a bit more specific.  I don't know if it matters, but it reads like your text is a special case of mine...

File: threatmodel.md (hunk 2 above)
Comment (mnm678) [reply threatmodel.md:15]: I updated my change. In some cases, someone might pull an old image by digest on purpose, and as long as the signature is still valid, this should be accepted, which is the subtlety I was trying to add. But yes, it still needs to use the latest tag mapping when tags are used.

File: threatmodel.md (hunk 3 above)
Comment (mnm678) [reply threatmodel.md:16]: I was trying to make it clear that DoS attacks are out of scope, but there might be a better way to word this.

File: threatmodel.md [hunk 7]
```diff
@@ -10,3 +10,15 @@ It is assumed that an attacker may perform one or more the following actions:
 
 While it is not always possible to protect against all scenarios, the system should to the extent possible mitigate and/or reduce the damage caused by a successful attack, detect the occurrence of an attack and notify appropriate parties, yet remain usable for parties operating the system.  Furthermore, the system should recover from successful attacks in a way that presents low operational overhead and risk to users.
 
+Attacker Goals:
+1. To have a party deploy a malicious artifact under the attacker's control.
+2. Trying to have a party install an outdated image.  For example, one with known security vulnerabilities.
+3. Making images unavailable for installation.
+4. Prevent a party from learning about updates to currently installed artifacts.
+5. Convince a party to download large amounts of data, such as signatures or metadata, that interfere with the party's system.
```
Comment (mnm678) [inline threatmodel.md:18]: This is covered by the size already listed in the registry, so is probably out of scope for Notary

**Comment by mnm678** [comment]
superceded by #76

**Pull request closed without merging**